File: backend/DTO/Response/ShipmentResponse.cs
using System.ComponentModel.DataAnnotations;

namespace backend.DTO.Response
{
    public class ShipmentResponse
    {
        [Required]
        public required int ShipmentId { get; set; }
        [Required]
        public required DateTime ShipmentDate { get; set; }
        [Required]
        public required string Address { get; set; }
        [Required]
        public required string City { get; set; }
        [Required]
        public required string State { get; set; }
        [Required]
        public required string Country { get; set; }
        [Required]
        public required string ZipCode { get; set; }
        [Required]
        public required string CustomerId { get; set; }
    }
}

File: frontend_admin/lib/model/response/shipment_response.dart
import 'package:equatable/equatable.dart';

class ShipmentResponse extends Equatable {
  const ShipmentResponse({
    this.shipmentId,
    this.shipmentDate,
    this.address,
    this.city,
    this.state,
    this.country,
    this.zipCode,
    this.customerId,
  });

  factory ShipmentResponse.fromJson(Map<String, dynamic> json) {
    return ShipmentResponse(
      shipmentId: json['shipmentId'] as int?,
      shipmentDate:
          json['shipmentDate'] != null
              ? DateTime.parse(json['shipmentDate'])
              : null,
      address: json['address'] as String?,
      city: json['city'] as String?,
      state: json['state'] as String?,
      country: json['country'] as String?,
      zipCode: json['zipCode'] as String?,
      customerId: json['customerId'],
    );
  }
  final int? shipmentId;
  final DateTime? shipmentDate;
  final String? address;
  final String? city;
  final String? state;
  final String? country;
  final String? zipCode;
  final String? customerId;

  Map<String, dynamic> toJson() {
    return {
      'shipmentId': shipmentId,
      'shipmentDate': shipmentDate?.toIso8601String(),
      'address': address,
      'city': city,
      'state': state,
      'country': country,
      'zipCode': zipCode,
      'customerId': customerId,
    };
  }

  @override
  List<Object?> get props => [
    shipmentId,
    shipmentDate,
    address,
    city,
    state,
    country,
    zipCode,
    customerId,
  ];
}
